webpack/components/AnsibleRoles/components/AnsibleRolesTableRowActionButton.js

```diff
@@ -0,0 +1,65 @@
+import React from 'react';
+import {
+  Dropdown,
+  DropdownToggle,
+  DropdownToggleAction,
+  DropdownItem,
+} from '@patternfly/react-core';
+import PropTypes from 'prop-types';
+import { translate as __, sprintf } from 'foremanReact/common/I18n';
+
+export const AnsibleRolesTableRowActionButton = props => {
+  const [isActionOpen, setIsActionOpen] = React.useState(false);
+  const onActionToggle = actionOpen => {
+    setIsActionOpen(actionOpen);
+  };
+
+  const dropdownItems = [
+    <DropdownItem
+      key="delete_action"
+      component="a"
+      data-method="delete"
+      href={`/ansible/ansible_roles/${props.ansibleRoleId}`}
+      data-confirm={sprintf(__('Delete %s?'), props.ansibleRoleName)}
+    >
+      {__('Delete')}
+    </DropdownItem>,
```

Comment:
AFAIK, the role that handles that permission is "Ansible Roles Manager", which is also required to view **Configure > Ansible > Roles**.
An unauthorized user is prevented from seeing the whole page.

Reply:
That's correct. However, you can set your own specific permissions. We need to cover both cases.

Reply:
Ah, I see... In that case that is indeed an issue. The problem is that permission handling capabilities are severely lacking on the React side.
It seems to me that this may only be accomplished with workarounds for now. See these PRs by kmalyjur:
https://github.com/theforeman/foreman_remote_execution/pull/842/files
https://github.com/theforeman/foreman/pull/9902

While that functionality has been merged, I'd be inclined to wait a bit before taking that approach because of this pending RFC:
https://community.theforeman.org/t/handling-permissions-whats-our-approach/36858